### PR TITLE
add path.Clean to user input paths

### PR DESCRIPTION
**Pull request by ahadda5**
**What type of PR is this?**

Bug fix 

**What does this PR do? Why is it needed?**
To prevent user inputted path from causing unintended security issues this PR adds path.Clean on every accepted user input.
The files are related to validators , where users can specify the wallet/data dir paths for accounts and db migration as well as for slashing export and import. 
The keystore tools takes in user-inputted paths as well. 
A few beacon-chain module's files also requires sanitization(path.clean) as well. 

**Which issues(s) does this PR fix?**

Fixes #
#9333 

**Other notes for review**
This PR does not deal with integrating [gosec ](https://github.com/securego/gosec)  with prysm's CI/PR checks .  That will be dealt with separately deciding on which rules in addition to G304.

### CODE REVIEW

**Review by rauljordan — DISMISSED** [review]

**Comment by rauljordan** [comment]
Looks like this causes a test failure which could be concerning. Maybe we can check why it led to that

**Comment by prestonvanloon** [comment]
Thanks for this PR! This doesn't fully resolve G304, but I don't see any gosec recommendations for path.Clean being reported.

[gosec_G304_results.txt](https://github.com/prysmaticlabs/prysm/files/6951002/results.txt)

**Review by prestonvanloon — CHANGES_REQUESTED** [review]
Please resolve all instances of G304 using filepath.Clean to mark #9333 as resolved.

**Comment by ahadda5** [comment]
I'll look at it first thing tomorrow morning.

File: beacon-chain/db/restore.go [hunk 1]
```diff
@@ -20,6 +20,9 @@ const dbExistsYesNoPrompt = "A database file already exists in the target direct
 func Restore(cliCtx *cli.Context) error {
 	sourceFile := cliCtx.String(cmd.RestoreSourceFileFlag.Name)
 	targetDir := cliCtx.String(cmd.RestoreTargetDirFlag.Name)
+	// Sanitize user input paths
```
Comment (terencechain) [inline beacon-chain/db/restore.go:23]: I will argue that we dont need these comments but rather rename function to `path.Sanitize()`. Much cleaner

Comment (rkapka) [reply beacon-chain/db/restore.go:23]: Good idea, although it requires changes to the standard Go library :-)

File: beacon-chain/node/registration/p2p.go [hunk 2]
```diff
@@ -28,6 +28,8 @@ func P2PPreregistration(cliCtx *cli.Context) (bootstrapNodeAddrs []string, dataD
 	}
 
 	dataDir = cliCtx.String(cmd.DataDirFlag.Name)
+	// Sanitize user input paths
+	dataDir = filepath.Clean(dataDir)
```
Comment (ahadda5) [inline beacon-chain/node/registration/p2p.go:32]: This causes the error. Clean replaces "" with a "."   so clean after the if.

**Comment by ahadda5** [comment]
> Thanks for this PR! This doesn't fully resolve G304, but I don't see any gosec recommendations for path.Clean being reported.
> 
> [gosec_G304_results.txt](https://github.com/prysmaticlabs/prysm/files/6951002/results.txt)
> 
> Edit: I looked more into this. G304 is resolved by path.Clean

I was debating whether to add it to all of them.  Some i missed and i'm fixing but others like the paths that already gets called by our own fileutil.ExpandPath uses path.Clean. I guess gosec is not advanced to see through the call stack. 
Also many variables listed are  non-user defined e.g  in boot_node.go gosec is asking us to clean non-user defined/inputed values like os.stdOutFile which i'm not sure is a security threat.  

I will add where necessary and outline what was not done on purpose like the two examples above unless you tell me to just add it everyone gosec complains?

**Comment by ahadda5** [comment]
Alright so here is the list of the files that did not make the cut with the reasoning. Why add code if we do not need it, otherwise I  can simply add @prestonvanloon 

Results:

No need , it uses prysm's Expanded func before which calls Clean.
```
[/home/ahaddad/prysm/validator/accounts/wallet/wallet.go:120] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    119: 	}
  > 120: 	f, err := os.Open(expanded)
    121: 	if err != nil {

```


No need,  the keysDir cleaned before the func call. 
```
[/home/ahaddad/prysm/validator/accounts/accounts_import.go:293] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    292: func readKeystoreFile(_ context.Context, keystoreFilePath string) (*keymanager.Keystore, error) {
  > 293: 	keystoreBytes, err := ioutil.ReadFile(keystoreFilePath)
    294: 	if err != nil {
```


No need , uses prysm's ExpandPath before which calls Cleans.
```
[/home/ahaddad/prysm/validator/accounts/accounts_import.go:253] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    252: 	}
  > 253: 	privKeyHex, err := ioutil.ReadFile(fullPath)
    254: 	if err != nil {

```

No need , uses prysm's ExpandPath before which calls Cleans.
```
[/home/ahaddad/prysm/tools/genesis-state-gen/main.go:66] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    65: 		}
  > 66: 		inputJSON, err := os.Open(expanded)
    67: 		if err != nil {

```

No need  postSSZFilepath is not user defined. 
```
[/home/ahaddad/prysm/spectest/shared/phase0/sanity/block_processing.go:86] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    85: 
  > 86: 				postBeaconStateFile, err := ioutil.ReadFile(postSSZFilepath)
    87: 				require.NoError(t, err)

```

No need  postSSZFilepath is not user defined. 
```
[/home/ahaddad/prysm/spectest/shared/phase0/operations/helpers.go:61] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    60: 
  > 61: 		postBeaconStateFile, err := ioutil.ReadFile(postSSZFilepath)
    62: 		require.NoError(t, err)
```



No need  postSSZFilepath is not user defined. 
```
[/home/ahaddad/prysm/spectest/shared/phase0/operations/block_header.go:59] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    58: 
  > 59: 				postBeaconStateFile, err := ioutil.ReadFile(postSSZFilepath)
    60: 				require.NoError(t, err)

```

No need  postSSZFilepath is not user defined. 
```
[/home/ahaddad/prysm/spectest/shared/phase0/epoch_processing/helpers.go:53] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    52: 
  > 53: 		postBeaconStateFile, err := ioutil.ReadFile(postSSZFilepath)
    54: 		require.NoError(t, err)

```

No need  -uses ExpandPath before which calls Cleans.
```
[/home/ahaddad/prysm/shared/promptutil/prompt.go:135] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    134: 		}
  > 135: 		data, err := ioutil.ReadFile(passwordFilePath)
    136: 		if err != nil {

```

No need  -uses ExpandPath before which calls Cleans.
```
[/home/ahaddad/prysm/shared/fileutil/fileutil.go:200] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    199: 	}
  > 200: 	return ioutil.ReadFile(filePath)
    201: }
```


No need  -pre-deteremined file paths . Not user defined. 
```
[/home/ahaddad/prysm/shared/benchutil/pregen.go:82] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    81: 	}
  > 82: 	blockBytes, err := ioutil.ReadFile(path)
    83: 	if err != nil {
```


No need  -pre-deteremined file paths . Not user defined. 
```
[/home/ahaddad/prysm/shared/benchutil/pregen.go:65] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    64: 	}
  > 65: 	beaconBytes, err := ioutil.ReadFile(path)
    66: 	if err != nil {
```


No need  -pre-deteremined file paths . Not user defined. 
```
[/home/ahaddad/prysm/shared/benchutil/pregen.go:48] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    47: 	}
  > 48: 	beaconBytes, err := ioutil.ReadFile(path)
    49: 	if err != nil {
```

No need   does not take user inputs. uses stdOutFile
```
[/home/ahaddad/prysm/endtoend/components/boot_node.go:87] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    86: func enrFromLogFile(name string) (string, error) {
  > 87: 	byteContent, err := ioutil.ReadFile(name)
    88: 	if err != nil {
```

**Review by nisdas — COMMENTED** [review]
Probably my main concern with this PR would be, does it break any existing user behaviour ? it modifies all user provided directories

**Comment by ahadda5** [comment]
@nisdas, fully trust your judgement. 
I can add , thought, that fileutil.ExpandPath does more of change than the std path.Clean. It is used throughout the system on user provided directories including replacing tilde with home dir , expanding embedded env var and then calling Clean (e.g /a/b/../c ->  a/c) .

**Comment by rauljordan** [comment]
Closing after some discussion on Discord. Changing user input is risky and does not make sense to introduce the risk given these functions are called from a trusted party (the user)

**Pull request closed without merging**